src/lib.rs

```diff
@@ -2439,6 +2455,32 @@ impl Build {
         Ok(())
     }
 
+    fn add_inherited_rustflags(&self, cmd: &mut Tool) -> Result<(), Error> {
+        let env_os = match self.getenv("CARGO_ENCODED_RUSTFLAGS") {
+            Some(env) => env,
+            // No encoded RUSTFLAGS -> nothing to do
+            None => return Ok(()),
+        };
+        let mut cc_flags: Vec<OsString> = env_os
+            .to_string_lossy()
+            .split("\u{1f}")
+            // Strip prefixes from rustc flags
+            .flat_map(|flag| {
+                if flag == "-Z" || flag == "-C" {
+                    None
+                } else if flag.starts_with("-Z") || flag.starts_with("-C") {
+                    Some(&flag[2..])
+                } else {
+                    Some(flag)
+                }
+            })
+            .flat_map(|flag| rustc_to_cc_flag(flag, cmd.family))
+            .collect();
+
+        cmd.args.append(&mut cc_flags);
```

Comment:
```suggestion
        cmd.args.extend(
            env_os
                .to_string_lossy()
                .split("\u{1f}")
                // Strip prefixes from rustc flags
                .flat_map(|flag| {
                    if flag == "-Z" || flag == "-C" {
                        None
                    } else if flag.starts_with("-Z") || flag.starts_with("-C") {
                        Some(&flag[2..])
                    } else {
                        Some(flag)
                    }
                })
                .flat_map(|flag| rustc_to_cc_flag(flag, cmd.family))
         );
```

Reply:
It is very unfortunate that we have to convert it to a utf-8 string for processing...

On unix/wasm we can at least convert it to bytes.

It should also use shlex, if the feature/dep is enabled.

shlex supports both bytes and str, so it can be used in both unix/wasm and windows.

Reply:
> It should also use shlex

So first pass it through shlex then split like in the snippet above? Shlex doesn't split on `\u{1f}` so running CARGO_ENCODED_RUSTFLAGS through it just ends up with a Vec of size 1 with the only element being all of CARGO_ENCODED_RUSTFLAGS.

Also, the suggested change above does not compile. If I use extend I need to mutably borrow cmd for it, so I then can't immutably borrow cmd to get cmd.family when calling rustc_to_cc_flag

Reply:
> Shlex doesn't split on `\u{1f}` so running CARGO_ENCODED_RUSTFLAGS through it just ends up with a Vec of size 1 with the only element being all of CARGO_ENCODED_RUSTFLAGS.

Thanks, I thought it's the same as other flags, turns out I'm wrony.

> If I use extend I need to mutably borrow cmd for it, so I then can't immutably borrow cmd to get cmd.family when calling rustc_to_cc_flag

Sorry I didn't notice that.

`cmd.family` is copyable, so you can copy it out to a variable and use it.